### PR TITLE
reduce inrepoconfig cache size defaults

### PR DESCRIPTION
**Pull request by listx**
It has been observed in practice that the gerrit component, under high load (where many of the changes being processed are using inrepoconfig), can run out of memory before reaching the 1000 keys limit. In our internal instance we were hitting ~470 keys before getting OOMKilled.

/cc @airbornepony @timwangmusic @cjwagner

### CODE REVIEW

**Comment by listx** [comment]
Before we merge this let's also first test that using 400 in our internal instance is actually OK (we'll be testing cache key evictions for the first time, I think).

**Comment by listx** [comment]
Update: we're now testing with 200 as the default. Seems to be OK. Updated to use 200 in this PR also.

File: prow/flagutil/config/config.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func (o *ConfigOptions) AddFlags(fs *flag.FlagSet) {
 	fs.Var(&o.SupplementalProwConfigDirs, "supplemental-prow-config-dir", "An additional directory from which to load prow configs. Can be used for config sharding but only supports a subset of the config. The flag can be passed multiple times.")
 	fs.StringVar(&o.SupplementalProwConfigsFileNameSuffix, "supplemental-prow-configs-filename", "_prowconfig.yaml", "Suffix for additional prow configs. Only files with this name will be considered. Deprecated and mutually exclusive with --supplemental-prow-configs-filename-suffix")
 	fs.StringVar(&o.SupplementalProwConfigsFileNameSuffix, "supplemental-prow-configs-filename-suffix", "_prowconfig.yaml", "Suffix for additional prow configs. Only files with this name will be considered")
-	fs.IntVar(&o.InRepoConfigCacheSize, "in-repo-config-cache-size", 1000, "Cache size for ProwYAMLs read from in-repo configs. Each host receives its own cache.")
+	fs.IntVar(&o.InRepoConfigCacheSize, "in-repo-config-cache-size", 200, "Cache size for ProwYAMLs read from in-repo configs. Each host receives its own cache.")
```
Comment (cjwagner) [inline prow/flagutil/config/config.go:60]: "Each host receives its own cache."  Is this still true?

**Review by cjwagner — APPROVED** [review]
/hold

File: prow/flagutil/config/config.go (hunk 1 above)
Comment (listx) [reply prow/flagutil/config/config.go]: Done.

**Review by cjwagner — APPROVED** [review]
/unhold

**Pull request merged**